File: backend/src/CodeReviewAnalyzer.Application/Models/PullRequestStats.cs
namespace CodeReviewAnalyzer.Application.Models;

public class PullRequestStats
{
    public required string ExternalId { get; init; }

    public required string Title { get; init; }

    public required DateTime CreatedAt { get; init; }

    public required DateTime ClosedAt { get; init; }

    public DateTime? FirstCommentDate { get; init; }

    public int FirstCommentWaitingMinutes { get; init; }

    public int RevisionWaitingTimeMinutes { get; init; }

    public int MergeWaitingTimeMinutes { get; init; }

    public string? MergeMode { get; init; }

    public int FileCount { get; init; }

    public int ThreadCount { get; init; }
}

File: frontend/src/app/pages/service/pull-request-repository/models/pull-request-stats.ts
export interface PullRequestStats {
    externalId?: string;
    title?: string;
    createdAt?: string;
    closedAt?: string;
    firstCommentDate?: string;
    firstCommentWaitingMinutes?: number;
    revisionWaitingTimeMinutes?: number;
    mergeWaitingTimeMinutes?: number | null;
    mergeMode?: string;
    fileCount?: number;
    threadCount?: number;
}
